### PR TITLE
Update libmesh

### PR DESCRIPTION
**Pull request by lindsayad**
Summary of changes:
  - Make CompareDofObjectsByID available to all
  - Make use of std::condition_variable for _get_array
  - Update to TIMPI 1.8.2
  - Check processor id of point neighbors
  - Nanoflann update to 1.4.1
  - Update to TIMPI 1.8.1
  - Use get_unique_tag() for a few more MessageTags
  - Remove caching from FE<CLOUGH>
  - Higher precision when printing assert failures
  - resolve warning of expectedly unbinned entries.
  - Clear input data structure before using
  - Don't unnecessarily compute the Jacobian for ComputeResidualandJacobian

### CODE REVIEW

**Comment by dschwen** [comment]
I've an upcoming libmesh thing as well (update of nanoflann). Just FYI.

**Comment by lindsayad** [comment]
we'll wait on https://github.com/libMesh/libmesh/pull/3122

File: modules/doc/content/newsletter/2022_01.md [hunk 1]
```diff
@@ -32,6 +32,16 @@ moose-petsc              3.15.1             build_4
...
+- Check processor id of point neighbors
+- Nanoflann update to 1.4.1
+- Update to TIMPI 1.8.1
+- Use get_unique_tag() for a few more MessageTags
+- Remove caching from FE<CLOUGH>
+- Higher precision when printing assert failures
+- resolve warning of expectedly unbinned entries.
+- Clear input data structure before using
+- Don't unnecessarily compute the Jacobian for ComputeResidualandJacobian
+
```
Comment (roystgnr) [inline modules/doc/content/newsletter/2022_01.md:44]: We should probably filter out the less interesting entries here (nobody cares about some binning warning) and expand the more interesting ones.

**Comment by dschwen** [comment]
I'll put in a fix for Magpie.

**Comment by dschwen** [comment]
For the Griffin fail, @YaqiWang can you check if you define `PI` as a macro anywhere in your code? That's the only explanation I can see for this weird compile error.

**Comment by cticenhour** [comment]
Working on what's going on with the Freya segfault.

**Comment by roystgnr** [comment]
In theory we can bump up to libMesh master now and no longer see the Freya segfault.

**Comment by lindsayad** [comment]
Yea. I can do that. I would like to see a decision on https://github.com/libMesh/libmesh/pull/3128 though too

**Comment by lindsayad** [comment]
> For the Griffin fail, @YaqiWang can you check if you define PI as a macro anywhere in your code? That's the only explanation I can see for this weird compile error.

@dschwen this is indeed in one of their headers:
```c++
#ifndef PI
#define PI acos((double)-1.0)
#endif
```

**Comment by roystgnr** [comment]
Ha, so would I.  If Derek's too busy to double-check you then I'll review it myself, but the first phase of that is going to be "learn a lot more about threading" so don't expect the review in less than a day or two.  I just fixed a "breaks 0.01% of the time" sort of bug, I'm really scared about possibly adding another one, and that's exactly the sort of bug that subtle errors in threading code creates.

**Comment by lindsayad** [comment]
Well you should definitely feel confident that the code is safe before merging it ... but I would also raise the question of whether you feel the old code is safe ... helgrind would suggest that it prefers the new code

**Comment by lindsayad** [comment]
Created https://gitlab.software.inl.gov/idaholab/griffin/-/issues/360 for `PI`. I'll probably look into this now

**Comment by dschwen** [comment]
> @dschwen this is indeed in ~one~ six of their headers:

Damn, I'm good.

**Comment by lindsayad** [comment]
[MR up](https://gitlab.software.inl.gov/idaholab/griffin/-/merge_requests/380)

**Comment by roystgnr** [comment]
Go ahead and update without waiting on my performance PR; it doesn't give enough improvement yet, and the next phase of it may take some time.

**Comment by lindsayad** [comment]
I can go ahead and re-trigger testing. I'm wondering if we might be able to get https://github.com/libMesh/libmesh/pull/3130 in though. Also I am not in a huge rush to get this merged ... I originally needed this for the compute residual-and-Jacobian together PR but I've moved onto some other things for the time being.

File: modules/doc/content/newsletter/2022_01.md (hunk 1 above)
Comment (lindsayad) [reply modules/doc/content/newsletter/2022_01.md:44]: I find binning warnings fascinating 🤣

Comment (cticenhour) [reply modules/doc/content/newsletter/2022_01.md:44]: I'm honestly fine with having everything in there - these newsletters should be useful to everybody, regardless of their opinion on binning warnings 😉

**Comment by lindsayad** [comment]
@roystgnr I filtered the newsletter a little bit and added some more description to the PRs I know about. Please feel free to edit more

File: conda/libmesh/meta.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Do not use jinja templating (A physical change to this file is required to trigger a build)
-{% set build = 1 %}
+{% set build = 0 %}
 {% set strbuild = "build_" + build|string %}
-{% set version = "2021.12.15" %}
+{% set version = "2022.01.19" %}
```
Comment (milljm) [inline conda/libmesh/meta.yaml:4]: 👍

File: conda/mpich/conda_build_config.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 ## Any changes made will require additional changes to any item above that.
 
 moose_libmesh:
-  - moose-libmesh 2021.12.15 build_1
+  - moose-libmesh 2022.01.19 build_0
```
Comment (milljm) [inline conda/mpich/conda_build_config.yaml:5]: 👍

**Review by milljm — DISMISSED** [review]
version change looks good to me. Its about all that I can comment on.

File: modules/doc/content/newsletter/2022_01.md [hunk 4]
```diff
@@ -32,6 +32,23 @@ moose-petsc              3.15.1             build_4
 
 ## libMesh-level Changes
 
+- Make 1CompareDofObjectsByID` available to all. This is useful for building sets
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2022_01.md:35]:
```suggestion
- Make `CompareDofObjectsByID` available to all. This is useful for building sets
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
One edit on the newsletter. Otherwise thanks for the additional content!

**Review by cticenhour — APPROVED** [review]
Unless @roystgnr has any suggestions for the newsletter, I am OK with this.

**Review by roystgnr — APPROVED** [review]

**Pull request merged**